hazelcast-client/src/main/java/com/hazelcast/client/proxy/PartitionSpecificClientProxy.java

```diff
@@ -43,21 +41,28 @@ protected void onInitialize() {
         partitionId = getContext().getPartitionService().getPartitionId(partitionKey);
     }
 
-    protected ClientMessage invokeOnPartition(ClientMessage req) {
-        return invokeOnPartition(req, partitionId);
+    protected ClientMessage invokeOnPartition(ClientMessage request) {
+        return invokeOnPartition(request, partitionId);
+    }
+
+    protected <T> T invokeOnPartition(ClientMessage request, ClientMessageDecoder decoder) {
```

Comment:
It requires further cleaning up. The method call isn't the issue; but the exception handling could be a hornets nest.

My goal is to get rid of client side duplication